modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -542,60 +574,52 @@ public FlowConstruct getFlowConstruct() {
     };
   }
 
-  private class StartSourceCallback implements RetryCallback {
-
-    boolean restarting;
-    RestartContext restartContext;
-
-    StartSourceCallback(boolean restarting, RestartContext restartContext) {
-      this.restarting = restarting;
-      this.restartContext = restartContext;
-    }
-
-    @Override
-    public void doWork(RetryContext context) throws Exception {
+  private void doWork(boolean restarting, RestartContext restartContext, CompletableFuture<Void> future) {
+    try {
+      createSource(restarting);
+      initialiseIfNeeded(sourceAdapter);
+      if (restarting) {
+        sourceAdapter.finishRestart(restartContext);
+      }
+      sourceAdapter.start();
+      future.complete(null);
+    } catch (Exception e) {
       try {
-        createSource(restarting);
-        initialiseIfNeeded(sourceAdapter);
-        if (restarting) {
-          sourceAdapter.finishRestart(restartContext);
-        }
-        sourceAdapter.start();
-      } catch (Exception e) {
-        try {
-          // On connection exception, if the failed connection is present, it must be invalidated before stopping the source. This
-          // warranties that a possible call to connectionProvider.disconnect made on the onStop method of the source, does not
-          // affect the connection's invalidation
-          extractConnectionException(e).ifPresent(connectionException -> invalidateConnection(connectionException));
-          stopSource();
-        } catch (Exception eStop) {
-          e.addSuppressed(eStop);
-        }
-        try {
-          disposeSource();
-        } catch (Exception eDispose) {
-          e.addSuppressed(eDispose);
-        }
-        Throwable throwable = exceptionEnricherManager.process(e);
-        Optional<ConnectionException> connectionException = extractConnectionException(throwable);
-        if (connectionException.isPresent()) {
-          throwable = connectionException.get();
-        }
-        throw throwable instanceof Exception ? ((Exception) throwable) : new MuleRuntimeException(throwable);
+        // On connection exception, if the failed connection is present, it must be invalidated before stopping the source. This
+        // warranties that a possible call to connectionProvider.disconnect made on the onStop method of the source, does not
+        // affect the connection's invalidation
+        extractConnectionException(e).ifPresent(connectionException -> invalidateConnection(connectionException));
+        stopSource();
+      } catch (Exception eStop) {
+        e.addSuppressed(eStop);
+      }
+      try {
+        disposeSource();
+      } catch (Exception eDispose) {
+        e.addSuppressed(eDispose);
```

Comment:
why?

Reply:
I kept this method exactly as it was before (only added the future as a param) see above the deleted lines above for reference.